File: 009/Ruby/problem009.rb
class Integer

  def pythagorean_triplet
    (1..self).to_a.each do |a|
      (1..(self-a)).to_a.each do |b|
        c = self - a - b
        return a * b * c if a*a + b*b == c*c
      end
    end
  end

end

File: 009/Haskell/Problem009.hs
module Problem009 (pythagoreanTriplet) where

pythagoreanTriplet :: Integer -> Integer
pythagoreanTriplet n = 
  head [ a * b * c | a <- [1..n], b <- [1..(n-a)], c <- [n-a-b], a^2 + b^2 == c^2]
